forge/forge/compile.py

```diff
@@ -114,7 +115,7 @@ class CompileContext:
     microbatch_size: int
     microbatch_count: int
     inputs: Union[torch.Tensor, List[torch.Tensor]]
-    optimizer: Optional[torch.optim.Optimizer] = None
+    optimizer: Optional[Union[torch.optim.Optimizer, forge.optimizers.Optimizer]] = None
     training: bool = False
```

Comment:
Nit-picking, but do we need a training flag? 

If we provide loss or optimizer, it is expected to force training mode? Correct me if I'm missing something.

Ofc, this comment is not related to this change, just a question for further API design.

Reply:
A need for training (or similar) flag will be needed when we switch to two fwd passes. When doing validation during training, we will switch a flag to false (as in torch model.eval()), and turn it back on when training is resumed.

Reply:
Hmm. Valid point. Let's discuss this next week. I have a few unknowns from my side on how to do a seamless switch during training.